### ISSUE TITLE
Unoptimized engine swapping during hires fix

### ISSUE THREAD

**Issue by SirVeggie** [NVIDIA/Stable-Diffusion-WebUI-TensorRT#132]
Currently the extension will always try to swap to the most fitting engine available for best performance. But it only looks at the generation happening in a vacuum. Let's say I have two engines, one dynamic for 512x512-1024x1024 and one static for 1024x1024. The generation will start with the dynamic engine at 512x512, the swap to the static engine in 1024x1024 hires fix phase since it's the more appropriate engine. It should not switch during hires fix whenever possible, as the switch causes a long delay, since the current engine can handle the target resolution.

In short:
1) Prevent engine swap during hires fix phase whenever possible as the swap loses more time than the gains.
2) Take hires resolution into account when choosing the engine at the start of the generation.

**Comment by SirVeggie** [comment]
Fantastic work on the extension by the way.

**Comment by SirVeggie** [comment]
Additionally/alternatively I would like to propose a setting "defer engine swap", so that it will keep using the same engine as long as possible, since there are other extensions like `adetailer` that perform automatic inpainting at the end of the generation.
At least attempting to prioritize swapping only in the beginning of a completely fresh generation.

**Comment by contentis** [comment]
The dev branch contains a fix for the engine selection heuristic. It will now try to select an engine that covers both the hi and lowres. If it can't it will inform the user.

Please not that this is still WIP and might have some bugs.

**Comment by SirVeggie** [comment]
Sometimes the `No valid profile found` error appears even when the generation values are still valid. One time this happened, just generating a couple images while setting unet to `none` and then switching back fixed it without touching the generation params at all.

I'll close this now since it's a duplicate.
